### ISSUE TITLE
Heavy performance hit when many items in a dictionary of a workflow task

### ISSUE THREAD

**Issue by jluethi** [fractal-analytics-platform/fractal-web#322]
I just observed that the performance of workflow task editing gets very bad when there are (reasonably) many entries for allowed channels in the Create OME-Zarr (multiplexing) task.

Specifically, the user has 6 cycles (6 keys in the dictionary for allowed channels) and each entry has 4 channel objects. Now, just loading the workflow page makes the browser unresponsive, it can take 5-10s (sometimes with a "browser unresponsive" showing up).

Let's evaluate the performance here and see if this can be reproduced. I see the same by logging in as that user on the UZH deployment.
Here is the example workflow
[workflow-export-4i per well workflow-1698072006544.json](https://github.com/fractal-analytics-platform/fractal-web/files/13072234/workflow-export-4i.per.well.workflow-1698072006544.json)

**Comment by tcompa** [comment]
A couple of sanity checks (cc @zonia3000):

1. After the new playwright PR, are we now running coverage tools also in non-dev environments (e.g. when we do something like `npm run build; node build`)?
2. If so, can we disable it? (unless it's known that it has no impact on performance)

I cannot say whether this is related to the current issue, but I'm asking because the current `npm run dev` logs includes lines like
```
vite:istanbul> Sourcemaps was automatically enabled for code coverage to be accurate.
 To hide this message set build.sourcemap to true, 'inline' or 'hidden'.
```
and I was wondering if this has implications re: performance.

**Comment by jluethi** [comment]
Ok, that's also interesting to check. The performance issue I observed was specific to the one workflow task page with the heavy load, the other workflow task pages were fine

**Comment by tcompa** [comment]
Another for-the-record remark (related, but not necessarily relevant for the current issue):

I think we should proceed and implement https://github.com/fractal-analytics-platform/fractal-web/issues/161: instead of only using some JSON Schema examples, we should (also) use the actual ones from fractal-tasks-core.

Once this is in-place, we can also add a benchmark of a couple of "complex" situations (e.g. the create-ome-zarr-multiplexing JSON Schema, with a somewhat large object of arguments to validate). For the moment I'm rather thinking about by-hand benchmarks (measure the time a certain set of operations take, and then check that it's below some threshold) and not about formal benchmarks (_à la_ https://github.com/tinylibs/tinybench); but we can of course re-discuss it.

**Comment by zonia3000** [comment]
> 
> ```
> vite:istanbul> Sourcemaps was automatically enabled for code coverage to be accurate.
>  To hide this message set build.sourcemap to true, 'inline' or 'hidden'.
> ```

@tcompa, I did a test with the file provided by @jluethi and an older fractal-web version seeing that the performance issue was there also before we introduced the tests. The, checking with the profiler revealed the cause of the issue. More about it on #324.

In any case that sourcemaps message is a bit weird and I can check also it.

**Comment by tcompa** [comment]
Closed with #324 (see discussion in the PR for further details on the number of popovers).
